Ludown/lib/parseFileContents.js

```diff
@@ -822,5 +792,31 @@ const addItemIfNotPresent = function(collection, type, value) {
         collection[type].push(itemObj);
     }  
 };
-
+/**
+ * Helper function to add an item to collection if it does not exist
+ * @param {object} collection contents of the current collection
+ * @param {LUISObjNameEnum} type item type
+ * @param {object} value value of the current item to examine and add
+ * @param {string []} roles possible roles to add to the item
+ * @returns {void} nothing
+ */
+const addItemOrRoleIfNotPresent = function(collection, type, value, roles) {
+    let existingItem = collection[type].filter(item => {return item.name == value});
+    if(existingItem.length !== 0) {
+        // see if the role exists
+        roles.forEach(role => {
+            if(!existingItem[0].roles.includes(role)) existingItem[0].roles.push(role);
```

Comment:
`Array.includes()` is technically a loop. Generally speaking, its a good idea to avoid nested loops when possible. Consider creating  a map of roles by id or name then check agains that in the `forEach()`
For example:
```js
const roleMap = existingItem[0].roles.reduce((map, role) => (map[role] = true, map), {});
roles.forEach(role => {
  if (!roleMap[role]) {
    existingItem[0].roles.push(role);
  }
})
```
This way you only loop twice instead of n^2